cli/src/main/java/org/commonjava/maven/ext/manip/Cli.java

```diff
@@ -0,0 +1,188 @@
+            System.err.println( "Unable to start Cli subsystem" );
+            System.exit( 1 );
+        }
+
+        pmeCli.run( args );
+    }
+
+    private void run( String[] args )
+    {
+        Options options = new Options();
+        options.addOption( "h", false, "Print this help message." );
+        options.addOption( Option.builder( "f" )
+                                 .longOpt( "file" )
+                                 .hasArgs()
+                                 .numberOfArgs( 1 )
+                                 .desc( "POM file" )
+                                 .build() );
+        options.addOption( Option.builder( "D" )
+                                 .hasArgs()
+                                 .numberOfArgs( 2 )
+                                 .valueSeparator( '=' )
+                                 .desc( "Java Properties" )
+                                 .build() );
+
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = null;
+        try
+        {
+            cmd = parser.parse( options, args );
+        }
+        catch ( ParseException e )
+        {
+            logger.debug( "Caught problem parsing ", e );
+            System.err.println( e.getMessage() );
+
+            HelpFormatter formatter = new HelpFormatter();
+            formatter.printHelp( "...", options );
+            System.exit( 1 );
+
+        }
+
+        System.out.println( "### Options " + parser.toString() );
+        System.out.println( "### Args " + Arrays.toString( cmd.getOptions() ) );
+        System.out.println( "### Args " + Arrays.toString( cmd.getArgs() ) );
+        System.out.println( "### Args " + cmd.getArgList() );
+        System.out.println( "### Properties " + cmd.getOptionProperties( "D" ) );
+
+        if (cmd.hasOption( 'D' ))
+        {
+            userProps = cmd.getOptionProperties( "D" );
+        }
+        if (cmd.hasOption( 'f' ))
+        {
+            target = new File (cmd.getOptionValue( 'f' ));
+        }
+
+        createSession(target);
+
+        if ( !session.isEnabled() )
+        {
+            logger.info( "Manipulation engine disabled via command-line option" );
+            return;
+        }
+        if ( !target.exists() )
+        {
+            logger.info( "Manipulation engine disabled. No project found." );
+            return;
+        }
+        else if ( new File( target, ManipulationManager.MARKER_FILE ).exists() )
+        {
+            logger.info( "Skipping manipulation as previous execution found." );
+            return;
+        }
+
+        try
+        {
+            manipulationManager.init( session );
+
+            logger.info( "### ee.getSession().getRequest().getPom() " + session.getTargetDir() + " and target "
+                                         + target );
+            logger.info ("### session.gettargetdir " + session.getPom() + " and target " + target);
+
+            manipulationManager.scanAndApply( session );
+        }
```

Comment:
Minimal CLI for first pass .... just enough to test basic manipulation functionality.